pyproject.toml

```diff
@@ -23,22 +23,32 @@ matplotlib = "^3.8.3"
 optional = true
 
 [tool.poetry.group.dev.dependencies]
-pyyaml = "^6.0.0"
+nbconvert = "^7.16.4"
+ipywidgets = "^8.1.3"
+deptry = "^0.17.0"
+
+[tool.poetry.group.docs]
+optional = true
+
+[tool.poetry.group.docs.dependencies]
 ipykernel = "^6.29.3"
+pillow = "^10.3.0"
+pyyaml = "^6.0.0"
 nbclient = "^0.10.0"
 nbformat = "^5.10.0"
```

Comment:
What are these used for? @CDCgov/multisignal-epi-inference-devs

Reply:
- Is `pillow` perhaps used somewhere in handling the output images bundled with the quarto-rendered `.rst` tutorials? @gvegayon. Might be worth trying a fresh render of the docs without it and seeing if anything breaks.
 
- I would guess `pyyaml` had something to do with loading of `yaml` formatted configs from, e.g. https://github.com/cdcgov/wastewater-informed-covid-forecasting/ But as above I think we can try removing it and seeing if it breaks things.

Reply:
I am also asking about `nbclient ` and `nbformat`.

Reply:
Don't remember exactly what was the pillow for, but `nbclient` and `nbformat` are required for rendering the quarto using the jupyter engine.

Reply:
Related: would be very nice to handle the dependency on the `quarto` CLI via `poetry` if possible. I don't think we currently do.

Note that this [official python interface to the `quarto` CLI](https://github.com/quarto-dev/quarto-python) lists `nbclient`, `nbformat`, and `pyyaml` as deps: 

https://github.com/quarto-dev/quarto-python/blob/4fcf29d33f8da520cdfd690d5dd53d556a21c09d/setup.py#L16

Reply:
@sbidari Based on this conversation, can you try removing `pillow`? Seems like we need to keep the others.

Reply:
There's a dependency for pillow somewhere, but I'm happy if that works :)!

Reply:
All the checks have passed even after removing pillow!